coderd/database/queries/users.sql

```diff
@@ -25,6 +25,25 @@ WHERE
 LIMIT
 	1;
 
+-- name: ValidateUserIDs :one
+WITH input AS (
+	SELECT
+		unnest(@user_ids::uuid[]) AS id
+)
+SELECT
+	array_agg(input.id)::uuid[] as invalid_user_ids,
+	COUNT(*) = 0 as ok
+FROM
+	-- Preserve rows where there is not a matching left (users) row for each
+	-- right (input) row...
+	users
+	RIGHT JOIN input ON users.id = input.id
+WHERE
+	-- ...so that we can retain exactly those rows where an input ID does not
+	-- match an existing user.
+	users.id IS NOT DISTINCT FROM NULL AND
+	users.deleted = false;
```

Comment:
Deleted users might be in the old ACLs when updating. Curious what happens if we delete someone, then try to update the ACL where they used to be in it.

Reply:
ooooh good point. maybe we only validate user ids that aren't `WorkspaceRoleDeleted`?

Reply:
because we need to handle this for groups too, and they won't be in the database at all.

Reply:
Good point on the groups. We should add a unit test for this and just make sure it all works.